heroku/resource_heroku_webhook.go

```
@@ -0,0 +1,214 @@
+package heroku
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	heroku "github.com/heroku/heroku-go/v5"
+)
+
+func resourceHerokuWebhook() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceHerokuWebhookCreate,
+		Read:   resourceHerokuWebhookRead,
+		Update: resourceHerokuWebhookUpdate,
+		Delete: resourceHerokuWebhookDelete,
+
+		Schema: map[string]*schema.Schema{
+			"app": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"level": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"url": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"include": &schema.Schema{
```

If there are only two acceptable values for `include`, you can add this:

```
ValidateFunc: validation.StringInSlice([]string{"x', "y"}, true),
```

---

I didn't know about this, awesome!

For `include` there are 11 valid event types, do you think it is worth enumerating them and forcing that restriction?

I think it would be nice to enforce that correctness, however it would mean requiring a code change if/when Heroku add or remove any additional event types.

---

I imagine that the Heroku API would return an error if the value is not allowable, but making this provider restrict values "at compile time" before making calls to Heroku API "at runtime" is an even better developer experience, failing faster.

It's simple enough to add more event types if/when Heroku releases them. Please follow @davidji99  suggestion and add this validation.